src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -83,6 +95,11 @@ public List<Setting<?>> getSettings() {
         );
     }
 
+    @Override
+    public Collection<Object> createComponents(SDKClient sdkClient, ClusterService clusterService, ThreadPool threadPool) {
+        return null;
```

Comment:
Not sure this is ever read anywhere, but isn't returning an empty collection better than null?

We don't need to override this if we implement in BaseExtension.

Reply:
Not extending from `AbstractAnomalyDetectorAction` anymore. So not needed, removed!